mmdet/models/test_time_augs/det_tta.py

```diff
@@ -0,0 +1,99 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+from typing import List
+
+import torch
+from mmcv.ops import batched_nms
+from mmengine.model import BaseTTAModel
+from mmengine.registry import MODELS
+from mmengine.structures import InstanceData
+
+from mmdet.structures import DetDataSample
+from mmdet.structures.bbox import bbox_flip
+
+
+def merge_aug_bboxes(aug_bboxes, aug_scores, img_metas):
+    """Merge augmented detection bboxes and scores.
+
+    Args:
+        aug_bboxes (list[Tensor]): shape (n, 4*#class)
+        aug_scores (list[Tensor] or None): shape (n, #class)
+    Returns:
+        tuple[Tensor]: ``bboxes`` with shape (n,4), where
+        4 represent (tl_x, tl_y, br_x, br_y)
+        and ``scores`` with shape (n,).
+    """
+    recovered_bboxes = []
+    for bboxes, img_info in zip(aug_bboxes, img_metas):
+        ori_shape = img_info['ori_shape']
+        flip = img_info['flip']
+        flip_direction = img_info['flip_direction']
+        if flip:
+            bboxes = bbox_flip(
+                bboxes=bboxes, img_shape=ori_shape, direction=flip_direction)
+        recovered_bboxes.append(bboxes)
+    bboxes = torch.cat(recovered_bboxes, dim=0)
+    if aug_scores is None:
+        return bboxes
+    else:
+        scores = torch.cat(aug_scores, dim=0)
+        return bboxes, scores
+
+
+@MODELS.register_module()
+class DetTTAModel(BaseTTAModel):
```

Comment:
Does the instance seg task require a new TTAModel?

Reply:
This PR only cover the TTA of object detection.